util/builtin_operators.go

```diff
@@ -331,7 +332,7 @@ func GlobMatchFunc(args ...interface{}) (interface{}, error) {
 
 // GenerateGFunction is the factory method of the g(_, _[, _]) function.
 func GenerateGFunction(rm rbac.RoleManager) govaluate.ExpressionFunction {
-	memorized := map[string]bool{}
+	memorized := sync.Map{}
```

Comment:
This was something that got me when I worked in this function earlier—at the time, the `g` function was called by govaluate purely serially, within the context of exactly one enforcement call. Is this something that is changed by the new expression caching?

Reply:
@timconinx

Reply:
@thetorpedodog 
I did change it, because the caching lead to concurrency problems in the `g` function. I discussed it in the related issue ([direct link](https://github.com/casbin/casbin/issues/1028#issuecomment-1153980413))

I'm not a real fan of replacing string bool maps by syncMaps, so if you see a better solution, then by all means